src/services/smart-contracts/actions/core.js

```diff
@@ -106,7 +111,134 @@ function getReadyCampaign(campaign, identity, Dai) {
 	return newCampaign
 }
 
-export async function openChannel({ campaign, account }) {
+async function getChannelsWithBalance({ identityAddr, requester, STATES }) {
+	const channels = await requester
+		.fetch({
+			route: '/campaigns',
+			method: 'GET',
+			queryParams: {
+				status: STATES.join(','),
+			},
+		})
+		.then(res => res.json())
+
+	return Promise.all(
+		channels
+			.map(channel => {
+				const { lastApprovedBalances } = channel.status
+				if (lastApprovedBalances) {
+					const allLeafs = Object.keys(lastApprovedBalances).map(k =>
+						Channel.getBalanceLeaf(k, lastApprovedBalances[k])
+					)
+					const mTree = new MerkleTree(allLeafs)
+					return { lastApprovedBalances, mTree, channel }
+				} else {
+					return { lastApprovedBalances: null, mTree: null, channel }
+				}
+			})
+			.filter(({ channel, lastApprovedBalances }) => {
+				if (channel.status.name === 'Expired') {
+					return channel.creator === identityAddr
+				}
+				return lastApprovedBalances && !!lastApprovedBalances[identityAddr]
+			})
+			.map(({ channel, lastApprovedBalances }) => ({
+				channel,
+				balance: lastApprovedBalances[identityAddr],
+			}))
+			.sort((c1, c2) => {
+				// Sorting by most balance so we can get top N needed using amountToSweep so we send as few transactions as possible
```

Comment:
no, you have to sort by `outstanding`; and of course, calculate `outstanding` in the beginning

outstanding, for expired channels, is `channel.deposit - core.withdrawn(channel.id)` and for non-expired channels it's `lastApprovedBalances[identityAddr] - core.withdrawPerUser(channel.id, identityAddr)`